tsl/test/expected/compression_job.out

```diff
@@ -0,0 +1,62 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+---
+-- Test the execution of two compression jobs in parallel
+---
```

Comment:
Better to implement as a isolation test.

Reply:
@mkindahl, @gayyappan Thanks for the advice, I have migrated the regression test to an isolation test.

Reply:
Have you checked that the isolation test fails without your fix?

Reply:
Summary of our discussion and why I asked if the test passed without your fix.

it looks like you're running the two sessions in a serial fashion since you do not have any waiting for locks. In that case, you would not see the error when you run the current version of the tests.

It is possible to make a step to immediately start waiting by adding a `*` (see the [README](https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/test/isolation/README)), but in that case the test would still rely on the relative speed of the two processes.

I think it is necessary to either explicitly lock some late lock to force them to wait and block on the locks in the correct manner, or add a debug waitpoint to the internals of the compression procedure so that you can block waiting for the locks there.

Reply:
Thanks for your explanation of the isolation tests. I have added two `DEBUG_WAITPOINTs` in the code to introduce a lock for chunk compression and for chunk decompression. The isolation test now has the required parallelism and I can reproduce the problem in the test. However, the test is unstable and I am currently working on stabilizing it.